### ISSUE TITLE
Workload Identity - Add Support for Custom KSA 

### ISSUE THREAD

**Issue by coding-trees** [triggermesh/triggermesh#1371]
Hello,

From my understanding in regards to Google Cloud services, the way that Workload Identity is implemented (https://github.com/triggermesh/triggermesh/pull/1223)  is it takes the name from `gcpServiceAccount` field and creates a KSA with the same name. Unfortunately for us, our company has varied naming conventions and our gcpServiceAccount naming convention is different from our kubernetes service account naming convention. 

If I have a workload identity set up where gcpServiceAccount is `test-gcp-134-svc` and the kubernetes service account is `ksa-test-abc-svc` the Workload Identity will not work as it will create a KSA named `test-gcp-134-svc` instead of `ksa-test-abc-svc`

Could an optional field be added that will take in the name of the KSA to create to allow for such scenarios?

Maybe added a `kubernetesServiceAccount` field to the [GoogleCloudSourceAuth](https://docs.triggermesh.io/1.24/reference/sources/#sources.triggermesh.io/v1alpha1.GoogleCloudSourceAuth) that when set will create the KSA with the name set in that field, otherwise create the KSA from the `gcpServiceAccount` field.

```yaml
auth:
  description:
  type: object
  properties:
    serviceAccountKey:
      description: Service account key used to authenticate the event source and allow it to interact with Google Cloud APIs. Only the JSON format is supported.
      type: object
      properties:
        value:
          description: Literal value of the service account key.
          type: string
        valueFromSecret:
          description: A reference to a Kubernetes Secret object containing the service account key.
          type: object
          properties:
            name:
              type: string
            key:
              type: string
          required:
            - name
            - key
      oneOf:
        - required: [value]
        - required: [valueFromSecret]
    gcpServiceAccount:
      description: GCP Service account name to impersonate Identity and Access Management (IAM) service accounts to access Google Cloud services.
      pattern: ^[a-z][a-z0-9-]+@[a-z0-9-]+\.[a-z]+\.[a-z]{2,63}$
      type: string
    kubernetesServiceAccount:
      description: Kubernetes Service account name bound to the gcpServiceAccount to act as an IAM service account. Optional, by default the kubernetes service account will be created with the same names as gcpServiceAccount.
      type: string
```

I can help contribute and work on adding this feature/change but might need some guidance. My initial thought was to add a string `ServiceAccountName` property to [ServiceAccountProvider](https://github.com/triggermesh/triggermesh/blob/main/pkg/apis/common/v1alpha1/object_interfaces.go#L86-L89) and set it to `rcl.GetName()` unless `s.Spec.Auth.kubernetesServiceAccount` is set. 

Thanks!

**Comment by tzununbekov** [comment]
Hi @coding-trees, thanks for filing the issue! 
You get the logic behind the GSA/KSA reconciliation almost right - the component's KSA is not the same as provided GSA: KSA name depends on the components configuration/requirements (shared SA or per-instance SA, [here](https://github.com/triggermesh/triggermesh/blob/v1.24.1/pkg/reconciler/adapter.go#L69-L85)).
Overall, the described case makes sense to me and I agree that we could add an optional `kubernetesServiceAccount` parameter to GCP CRDs. Regarding the implementation, before changing the [ServiceAccountProvider](https://github.com/triggermesh/triggermesh/blob/main/pkg/apis/common/v1alpha1/object_interfaces.go#L86-L89) interface, I'd look at the [`ServiceAccountOptions`](https://github.com/triggermesh/triggermesh/blob/v1.24.1/pkg/apis/sources/v1alpha1/googlecloudpubsub_lifecycle.go#L72-L82) to see if it is possible to add the name override ServiceAccountOption there. I'm not saying it is the only way it should be implemented, but it feels less intrusive than changing the interface. Let me know if this works for you. Otherwise, I can also give it a try next week.

**Comment by coding-trees** [comment]
Hi @tzununbekov,

Thanks for the suggestion, I made an attempt to implement the changes (please see PR https://github.com/triggermesh/triggermesh/pull/1372). Please let me know if I'm way off or on the right track, and of any and all changes that are needed.

**Comment by tzununbekov** [comment]
Resolved in #1372
